### PR TITLE
Add custom logic to truncate extra decimal digits

### LINKED ISSUE #29: Some amounts change while typing
An example:

```
val amount = "515.809"
currencyEditText.setText(amount)
```

A text visible in a field is expected to be `515.809` and yet it becomes `515.808`. 
I get the same results when trying to enter `515.809` manually.

### PR DESCRIPTION
**Pull request by efguydan**
There is an issue with using RoundingMode.DOWN to truncate extra decimal digits (to prevent RoundingMode.UP)
This Pull Request adds a custom helper method to truncate those extra decimal digits; tested extensively with Unit Tests.

Fixes #29

### CODE REVIEW

**Comment by rasheedsulayman** [comment]
👀

File: library/src/test/java/com/cottacush/android/currencyedittext/CurrencyRounderTest.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cottacush.android.currencyedittext
+
+import org.junit.Assert
+import org.junit.Test
+
+class CurrencyRounderTest {
```
Comment (rasheedsulayman) [inline library/src/test/java/com/cottacush/android/currencyedittext/CurrencyRounderTest.kt]: This test class seems too long. Can you check if you can reduce it this [parameterized-tests](https://github.com/junit-team/junit4/wiki/parameterized-tests)?

**Comment by oyeleke** [comment]
🌚

**Comment by oyeleke** [comment]
Just here thinking out aloud truncating can be a bit dangerous in digits especially when it comes to money.

File: library/src/test/java/com/cottacush/android/currencyedittext/CurrencyRounderTest.kt (hunk 1 above)
Comment (efguydan) [reply library/src/test/java/com/cottacush/android/currencyedittext/CurrencyRounderTest.kt]: checking this out.

**Comment by efguydan** [comment]
Very true @oyeleke. But in this scenario, we're trying to truncate extra decimal digits. 
i.e. if a dev says they want a max of 2 decimal digits, and the user wants to enter 192.568, the last 8 is truncated to display a value of 192.56 in the EditText. 
Ideally, Rounding down should provide the same result, but for some reason, it's giving wrong result for some values for some devices.

**Comment by oyeleke** [comment]
hmm yeah, I get I am just worried about modifying the value of money, cos of accounting and all of that. is it possible we just bounce the user, such that if a max decimal digit is selected a user isn't allowed to enter more digits, protects the integrity of the value, and ensures you are safe when risk and compliance comes over 🥲

**Comment by efguydan** [comment]
😄
Yeah, so that's what this "truncating" is trying to simulate. 
The user experience would be like the EditText isn't accepting decimal inputs anymore when they already have the max number of decimal digits. 
For example, say the max decimal digit allowed is 2, 
and the edit text is displaying 123.56
if the user were to click a number on their keyboard, say 8,
the watcher would receive an `onTextChanged` event with the value "123.568". However, it removes that last "8" and keeps the Edittext's value at "123.56".

It's basically a Rounding Down operation.

**Review by rasheedsulayman — APPROVED** [review]

**Pull request merged**